### DISCUSSION TITLE
Uniref30

### DISCUSSION THREAD

**Discussion by rukibuki** [KosinskiLab/AlphaPulldown#314] [General]
we have been using an old version of alphapulldown, so recently I tried installing the newest version (using pip)
but when I ran it, I got an error like this:
E0415 11:37:05.052870 140405823498048 hhblits.py:82] Could not find HHBlits database /scratch/alphafold_data_new/uniref30/UniRef30_2023_02

and we do not have that file in our AF database, our uniref30 folder contains these files:
-rw-r--r-- 1 524602 9100 6.8G Jul 19  2021 UniRef30_2021_03_cs219.ffdata
-rw-r--r-- 1 524602 9100 684M Jul 19  2021 UniRef30_2021_03_cs219.ffindex
-rw-r--r-- 1 524602 9100 160G Jul 19  2021 UniRef30_2021_03_a3m.ffdata
-rw-r--r-- 1 524602 9100 758M Jul 19  2021 UniRef30_2021_03_a3m.ffindex
-rw-r--r-- 1 524602 9100  22M Jul 19  2021 UniRef30_2021_03_hhm.ffindex
-rw-r--r-- 1 524602 9100  38G Jul 19  2021 UniRef30_2021_03_hhm.ffdata
-rw-r--r-- 1 524602 9100  379 Jul 19  2021 UniRef30_2021_03.md5sums

so all of them are 2021 files, which I found was odd, so I checked the uniref30_download script from AF: [link](https://github.com/google-eepmind/alphafold/blob/6a3af1adb3bbbc53562da100e3819b2fc882f915/scripts/download_uniref30.sh
)
that is part of the download_all_data.sh script from AF.  And it is indeed set to get the 2021 files that we have in our database.

So is our database correct and APD calls the wrong uniref30 files or is the uniref30 download script at AF wrong, which also means our AF database is outdated? I am betting on the last, considering the uniref30 download script, looks like it has not been updated for 2 years. But that would mean ppl are not using the download_all_data.sh script I guess, which I find surprising?

**Comment by dingquanyu** [comment]
Hi Ruki,

You're right about the version and now we should use 2023 version of the database. Could you try updating your uniref database and then run again? The same problem also was discussed here: #310 alternatively you could stick with the older version of AlphaPulldown and keep using 2021 version of the database.

Yours
Dingquan

**Reply by rukibuki** [reply]
Hi Dingquan,

Sorry I missed the 310 issue! I only looked in discussions, my bad.
The problem with the old version we are using, is that the "model_to_relax" flag is not supported in that version (plus that  analysis pipeline is not really working for us). I believe there is amber_relaxer flag, but that seems to relax all of the models and not just the best, which is an option in the newer version, AFAIK. 

I think we will have to update our AF database then.

**Reply by dingquanyu** [reply]
Hi @rukibuki 

Just so you know that relaxation does not work at all at the moment, as seen in the discussion here: https://github.com/google-deepmind/alphafold/issues/923 We're waiting for the solutions from Deepmind right now so I wouldn't recommend changing that flag by yourself

Yours
Dingquan

**Reply by rukibuki** [reply]
Thanks Dingquanyu, I had actually already played with it and never got it to work, but just brushed it off as being due to us running an old version of APD. But at least it is good to know it is a more general error.

**Comment by jkosinski** [comment]
@dingquanyu, don't we have command line arguments to change the specific database versions, the same as run_alphafold.py has? (--uniref30_database_path)
